core/arch/aarch64/encode.c

```diff
@@ -150,8 +150,8 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     if (instr_is_label(instr))
         return copy_pc;
 
-    /* First, handle the already-encoded instructions */
-    if (instr_raw_bits_valid(instr)) {
+    /* First, handle the already-encoded instructions. OP_ldstex is always relocatable. */
+    if (instr_raw_bits_valid(instr) || instr_get_opcode(instr) == OP_ldstex) {
```

Comment:
Why do we need this special check?  I would think we can just use the raw_bits_valid check and this should encode just like a regular Level 0 or Level 1 instr.